tools/shared.py

```diff
@@ -1655,6 +1655,10 @@ def js_optimizer(filename, passes, debug=False, extra_info=None, output_filename
     return ret
 
   @staticmethod
+  def eliminate_duplicate_funcs(filename):
+    duplicate_function_eliminator.eliminate_duplicate_funcs(filename)
```

Comment:
please do the import in this function - that way it won't slow down startup when it isn't used.